### PR TITLE
Build: remove unneeded dependencies & upgrades

### PR DESCRIPTION
**Pull request by ndelangen**
## What I did

- hopefully improve CI perf by removing dependencies we do not need/use

### CODE REVIEW

File: code/package.json [hunk 1]
```diff
@@ -279,44 +237,28 @@
     "jest-os-detection": "^1.3.1",
     "jest-serializer-html": "^7.1.0",
     "jest-watch-typeahead": "^2.2.1",
-    "lerna": "^3.22.1",
+    "lerna": "^6.4.0",
```
Comment (ndelangen) [inline code/package.json:240]: @shilman ☝️

File: code/addons/storyshots-core/stories/storyshot.enzyme.test.js [hunk 2]
```diff
@@ -1,7 +1,11 @@
 import path from 'path';
-import { mount } from 'enzyme';
+import { mount, configure } from 'enzyme';
+// @ts-expect-error (Converted from ts-ignore)
```
Comment (IanVS) [inline code/addons/storyshots-core/stories/storyshot.enzyme.test.js:3]: I know this is copy/paste from the init script, but I think we should install `@types/enzyme-adapter-react-16` instead of ignoring the error.

File: code/renderers/web-components/src/docs/web-components-properties.test.ts [hunk 3]
```diff
@@ -11,10 +11,14 @@ const inputRegExp = /^input\..*$/;
 const runWebComponentsAnalyzer = (inputPath: string) => {
   const { name: tmpDir, removeCallback } = tmp.dirSync();
   const customElementsFile = `${tmpDir}/custom-elements.json`;
-  spawnSync('yarn', ['wca', 'analyze', inputPath, '--outFile', customElementsFile], {
-    stdio: 'inherit',
-    shell: true,
-  });
+  spawnSync(
+    path.join(__dirname, '../../../../node_modules/.bin/wca'),
```
Comment (IanVS) [inline code/renderers/web-components/src/docs/web-components-properties.test.ts:15]: Why this change?  It feels a bit fragile to provide a direct path like this, though in reality it's probably fine.

File: code/ui/manager/src/settings/shortcuts.test.tsx [hunk 4]
```diff
@@ -1,175 +0,0 @@
...
-} as any;
-
-const makeActions = () => ({
-  setShortcut: jest.fn(),
-  restoreDefaultShortcut: jest.fn().mockImplementation((action) => shortcutKeys[action]),
-  restoreAllDefaultShortcuts: jest.fn().mockReturnValue(shortcutKeys),
-  onClose: jest.fn(),
-});
-
-describe('ShortcutsScreen', () => {
```
Comment (IanVS) [inline code/ui/manager/src/settings/shortcuts.test.tsx:26]: Why delete this test?  It doesn't seem related to dependency changes.

**Review by IanVS — COMMENTED** [review]
Nice!  I love me some good dependency cleanups.  

I have been meaning to ask, when do we need to add dependencies to the top-level package.json, and when do we not?  It seems like much of the cleanup you're doing is removing packages there.  I thought yarn monorepos should handle hoisting up packages to the top level automatically, so I would guess that only dev tools that need to run _at the top level_ need to be included, but it seems like there's much more.  Is that so that `yarn task` can work properly, or something?

File: code/ui/manager/src/settings/shortcuts.test.tsx (hunk 4 above)
Comment (ndelangen) [reply code/ui/manager/src/settings/shortcuts.test.tsx:26]: It's all enzyme, would love to get it replaced by something more modern, but removed it because it should already be covered (a bit) by chromatic anyway.

**Comment by ndelangen** [comment]
@IanVS regarding where dependencies should be located.

I'd like to have all devDependencies that are unique in the packages' devDependencies.
When they are shared, they should likely be moved to the scripts directory to get uniformity.

I'd like to optimize a bit for future splitting the monorepo apart more.

I think code/package.json should have 0 dependencies, except for maybe internal ones, that's a long term goal.

Very open to discussing if that's a good goal and on how to achieve it if it indeed is.!

File: code/renderers/web-components/src/docs/web-components-properties.test.ts (hunk 3 above)
Comment (ndelangen) [reply code/renderers/web-components/src/docs/web-components-properties.test.ts:15]: I'm not sure what exactly caused it to break, but jest/yarn couldn't find the binary anymore, and this fixed it.

I shortly tried to convert it to using the node API, but it seems the output of the nodeAPI is completely different..
I gave up after a while.

I also considered just checking in the generated code, and not running this generator at all anymore, but there's likely a reason this is done that way? Maybe it's to detect changes in the behavior of that package upstream? All seems pretty fragile already to me...

File: code/addons/storyshots-core/stories/storyshot.enzyme.test.js (hunk 2 above)
Comment (ndelangen) [reply code/addons/storyshots-core/stories/storyshot.enzyme.test.js:3]: I care extremely little about it, it's storyshots + an internal test + a test in combination with enzyme.

I don't want to spend more time on enzyme, I was considering remove this test as well, I'm not sure it's adding much to our true coverage...

**Review by valentinpalkovic — APPROVED** [review]

**Comment by ndelangen** [comment]
@shilman I'm going to hold off merging this, until you've given an 👌 sign, about the lerna upgrade.
